src/test/java/org/jenkinsci/plugins/envinject/EnvInjectJobPropertyTest.java

```diff
@@ -0,0 +1,134 @@
+        assertEquals("BAR", build.getEnvironment(TaskListener.NULL).get("FOO"));
+    }
+    
+    @Test
+    public void testVariableInjection_PropertyDisabled() throws Exception {   
+        FreeStyleProject project = j.createFreeStyleProject();
+        
+        EnvInjectJobProperty<FreeStyleProject> prop = new EnvInjectJobProperty<FreeStyleProject>();
+        prop.setInfo(new EnvInjectJobPropertyInfo(null, "FOO=BAR", null, null, null, false));
+        // prop.setOn(false); - It is default
+        project.addProperty(prop);
+        
+        FreeStyleBuild build = j.buildAndAssertSuccess(project);
+        assertNull("The plugin should not inject properties by default", build.getEnvironment(TaskListener.NULL).get("FOO"));
+    }
+    
+    @Test
+    public void testKeepBuildVariables_Default() throws Exception {   
+        FreeStyleProject project = j.createFreeStyleProject();
+        
+        EnvInjectJobProperty<FreeStyleProject> prop = new EnvInjectJobProperty<FreeStyleProject>();
+        prop.setOn(true);
+        prop.setInfo(new EnvInjectJobPropertyInfo(null, null, null, null, null, false));
+        // Default keepBuildVariables should be true
+        project.addProperty(prop);
+        
+        QueueTaskFuture<FreeStyleBuild> scheduled = project.scheduleBuild2(0, new Cause.UserIdCause(),
+                new ParametersAction(new StringParameterValue("PARAM", "Value")));
+        assertNotNull(scheduled);
+        FreeStyleBuild build = scheduled.get();
+        assertEquals(Result.SUCCESS, build.getResult());
+        assertEquals("Value", build.getEnvironment(TaskListener.NULL).get("PARAM"));
+    }
+    
+    @Test
+    @Ignore("The value is being actually contributed by other env contributors. The feature seems to be obsolete")
+    public void testKeepBuildVariables_False() throws Exception {   
+        FreeStyleProject project = j.createFreeStyleProject();
+        
+        EnvInjectJobProperty<FreeStyleProject> prop = new EnvInjectJobProperty<FreeStyleProject>();
+        prop.setOn(true);
+        prop.setInfo(new EnvInjectJobPropertyInfo(null, null, null, null, null, false));
+        prop.setKeepBuildVariables(false);
+        project.addProperty(prop);
+        
+        QueueTaskFuture<FreeStyleBuild> scheduled = project.scheduleBuild2(0, new Cause.UserIdCause(),
+                new ParametersAction(new StringParameterValue("PARAM", "Value")));
+        assertNotNull(scheduled);
+        FreeStyleBuild build = scheduled.get();
+        assertEquals(Result.SUCCESS, build.getResult());
+        assertNull("We expect that the PARAM is not specified", build.getEnvironment(TaskListener.NULL).get("PARAM"));
+    }
+    
+    @Test
+    @Ignore("It should not override vars accroding to the manual testing. But it does... "
+            + "Manual tests also show the wrong value in InjectedVarsAction")
+    public void testOverrideBuildParameters_Default() throws Exception {   
```

Comment:
@recena This is an issue I've discovered in https://github.com/jenkinsci/envinject-plugin/pull/56 . After an investigation it's not related to your PR